consensus/dummy/dynamic_fee_window.go

```diff
@@ -0,0 +1,88 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package dummy
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/coreth/params"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+var ErrDynamicFeeWindowInsufficientLength = errors.New("insufficient length for dynamic fee window")
+
+// DynamicFeeWindow is a window of the last [params.RollupWindow] seconds of gas
+// usage.
+//
+// Index 0 is the oldest entry, and [params.RollupWindow]-1 is the current
+// entry.
+type DynamicFeeWindow [params.RollupWindow]uint64
+
+func ParseDynamicFeeWindow(bytes []byte) (DynamicFeeWindow, error) {
+	if len(bytes) < params.DynamicFeeExtraDataSize {
+		return DynamicFeeWindow{}, fmt.Errorf("%w: expected at least %d bytes but got %d bytes",
+			ErrDynamicFeeWindowInsufficientLength,
+			params.DynamicFeeExtraDataSize,
+			len(bytes),
+		)
+	}
+
+	var window DynamicFeeWindow
+	for i := range window {
+		offset := i * wrappers.LongLen
+		window[i] = binary.BigEndian.Uint64(bytes[offset:])
+	}
+	return window, nil
+}
+
+// Add adds the amounts to the most recent entry in the window.
+//
+// If the most recent entry overflows, it is set to [math.MaxUint64].
+func (w *DynamicFeeWindow) Add(amounts ...uint64) {
+	const lastIndex uint = params.RollupWindow - 1
+	w[lastIndex] = add(w[lastIndex], amounts...)
+}
+
+// Shift removes the oldest amount entries from the window and adds amount new
```

Comment:
```suggestion
// Shift removes the oldest `amount` entries from the window and adds `amount` new
```

Or rename it to `n` (it's a short function so that's fine, and `n` is self-explanatory). Being an English word it was very confusing to read "removes the oldest amount" (valid sentence thus far) only to then realise that it's a variable.

https://en.wikipedia.org/wiki/Garden-path_sentence